tests/test_key.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+
+"""Test cases for "key.py". """
+
+import copy
+import unittest
+
+import securesystemslib.formats
+import securesystemslib.keys as KEYS
+from securesystemslib.exceptions import (
+    CryptoError,
+    FormatError,
+    UnsupportedAlgorithmError,
+)
+from securesystemslib.key import SSlibKey
+from securesystemslib.signer import SSlibSigner
+
+
+class TestSSlibKey(unittest.TestCase):
+    """SSlibKey Test Case."""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.rsakey_dict = KEYS.generate_rsa_key()
+        cls.ed25519key_dict = KEYS.generate_ed25519_key()
+        cls.ecdsakey_dict = KEYS.generate_ecdsa_key()
+        cls.DATA_STR = "SOME DATA REQUIRING AUTHENTICITY."
+        cls.DATA = securesystemslib.formats.encode_canonical(cls.DATA_STR).encode(
+            "utf-8"
+        )
+
+    def test_sslib_verify(self):
+        """Test to check verify method of key."""
+
+        dicts = [self.rsakey_dict, self.ecdsakey_dict, self.ed25519key_dict]
+        for scheme_dict in dicts:
+            sslib_signer = SSlibSigner(scheme_dict)
+            sig_obj = sslib_signer.sign(self.DATA)
+
+            # Verify signature
+            sslib_key = SSlibKey.from_securesystemslib_key(scheme_dict)
+            verified = sslib_key.verify(sig_obj, self.DATA)
+            self.assertTrue(verified, "Incorrect signature.")
+
+            # Test for invalid public key.
+            public = scheme_dict["keyval"]["public"]
+            scheme_dict["keyval"]["public"] = ""
+            sslib_key = SSlibKey.from_securesystemslib_key(scheme_dict)
+
+            with self.assertRaises(
+                (CryptoError, UnsupportedAlgorithmError, FormatError)
+            ):
+                sslib_key.verify(sig_obj, self.DATA)
+
+            scheme_dict["keyval"]["public"] = public
+
+    def test_sslib_serialization(self):
```

Comment:
Can you add one more test case: `Test for load and dump key_dict without "keyid".` as it's optional.

Reply:
Done